n8n/templates/service-main.yaml

```diff
@@ -37,3 +37,42 @@ spec:
     service: n8n
   ports:
     - port: 5678
+
+{{- if .Values.n8n.ingress.enabled }}
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  {{- with .Values.n8n.ingress.annotations }}
+  annotations:
```

Comment:
For the websockets to work reliably, I recommend adding [these annotations](https://github.com/n8n-io/n8n-eks-cluster/blob/main/n8n/templates/ingress.yaml#L5-L10)

Reply:
Hi @netroy,

These can be passed in `values.yaml` file.

My idea was to keep the templates clean, and everyone can provide their own annotations.

For example, I tested this Helm chart in the home lab with the Traefik ingress controller, and I used this:
```yaml
n8n:
  ingress:
    enabled: true
    className: traefik
    annotations:
      traefik.ingress.kubernetes.io/router.entrypoints: websecure
      traefik.ingress.kubernetes.io/router.tls: "true"
      traefik.ingress.kubernetes.io/router.tls.certresolver: le
    hosts:
      - host: n8n.balutoiu.com
        paths:
          - path: /
            pathType: Prefix

webhook:
  ingress:
    enabled: true
    className: traefik
    annotations:
      traefik.ingress.kubernetes.io/router.entrypoints: websecure
      traefik.ingress.kubernetes.io/router.tls: "true"
      traefik.ingress.kubernetes.io/router.tls.certresolver: le
    hosts:
      - host: n8n-webhook.balutoiu.com
        paths:
          - path: /
            pathType: Prefix
```

And it worked fine.

However, I believe it would be appropriate to have [these annotations](https://github.com/n8n-io/n8n-eks-cluster/blob/74c30b361d06100055aae5d53762d20d42f1c8df/n8n/templates/ingress.yaml#L5-L10) in the `values.yaml` embedded with the helm chart.

Also, I see that the ingress resource from [here](https://github.com/n8n-io/n8n-eks-cluster/blob/main/n8n/templates/ingress.yaml) is shared between the main service and the webhook service. That's a great idea!

I believe we can have the `templates/ingress.yaml` from [here](https://github.com/n8n-io/n8n-eks-cluster/blob/main/n8n/templates/ingress.yaml), and my only improvement would be keeping the template file generic, and adding values for ingress annotations / className (these depend on the ingress controller deployed in the Kubernetes cluster).

What do you think?

Reply:
Since the ingress class specific configuration is important working on n8n. could we instead wrap the annotations in checks like this?
```yaml
{{- if eq .Values.n8n.ingress.className "nginx" }}
```
That way the chart includes the best practices in the templates, and the values file could be much smaller.

Reply:
> Since the ingress class specific configuration is important working on n8n. could we instead wrap the annotations in checks like this?
> 
> ```yaml
> {{- if eq .Values.n8n.ingress.className "nginx" }}
> ```
> 
> That way the chart includes the best practices in the templates, and the values file could be much smaller.

I agree! Making the changes to update this PR.

Reply:
This is how my homelab n8n with traefik ingress looks now:
```yaml
ingress:
  enabled: true
  className: traefik
  annotations:
    traefik.ingress.kubernetes.io/router.entrypoints: websecure
    traefik.ingress.kubernetes.io/router.tls: "true"
    traefik.ingress.kubernetes.io/router.tls.certresolver: le
```

vs before:
```yaml
n8n:
  ingress:
    enabled: true
    className: traefik
    annotations:
      traefik.ingress.kubernetes.io/router.entrypoints: websecure
      traefik.ingress.kubernetes.io/router.tls: "true"
      traefik.ingress.kubernetes.io/router.tls.certresolver: le
    hosts:
      - host: n8n.balutoiu.com
        paths:
          - path: /
            pathType: Prefix

webhook:
  ingress:
    enabled: true
    className: traefik
    annotations:
      traefik.ingress.kubernetes.io/router.entrypoints: websecure
      traefik.ingress.kubernetes.io/router.tls: "true"
      traefik.ingress.kubernetes.io/router.tls.certresolver: le
    hosts:
      - host: n8n-webhook.balutoiu.com
        paths:
          - path: /
            pathType: Prefix
```

In my opinion, it is a lot cleaner with a combined ingress resource.

Reply:
Do you think it makes sense to add traefik specific entries in the values file, and move the annotations in the template?
so if `ingress.traefik.tls` is true the add the `router.entrypoints` and and `router.tls` annotations, and unless `ingress.traefik.tls.certresolver` is set to another value, that annotation could also default to let's encrypt.

I'm assuming that this is a fairly common setup, and it should be okay to support this in the templates just like with nginx specific annotations.

That said, I'm also fine leaving it this way.
Since we don't have a CI setup for this yet. I'd like to test this manually later just to be sure, and then we can merge it.

Thanks 💟

Reply:
> Do you think it makes sense to add traefik specific entries in the values file, and move the annotations in the template? so if `ingress.traefik.tls` is true the add the `router.entrypoints` and and `router.tls` annotations, and unless `ingress.traefik.tls.certresolver` is set to another value, that annotation could also default to let's encrypt.
> 
> I'm assuming that this is a fairly common setup, and it should be okay to support this in the templates just like with nginx specific annotations.

We could do this, but I believe it's cleaner the way it is this way now. Because these traefik annotations are specific to my homelab setup, and another person with Treaefik might have different annotations (that's why I wouldn't include them in templates).

However, I would still have these traefik annotations commented in the `values.yaml` file, for reference, like this:
```yaml
ingress:
  enabled: false
  className: nginx
  annotations: {}
  #
  # Traefik ingress example
  #
  # className: traefik
  # annotations:
  #   traefik.ingress.kubernetes.io/router.entrypoints: websecure
  #   traefik.ingress.kubernetes.io/router.tls: "true"
  #   traefik.ingress.kubernetes.io/router.tls.certresolver: le

```

This way, other people can use them as inspiration when configuring Traefik ingress with n8n.

> That said, I'm also fine leaving it this way. Since we don't have a CI setup for this yet. I'd like to test this manually later just to be sure, and then we can merge it.
> 
> Thanks 💟

No problem. Thank-you for the review!